addons/web_editor/static/src/js/common/utils.js

```diff
@@ -487,6 +503,89 @@ function _isMobileView(targetEl) {
+ * on the "image.data" registry.
+ * @param {Object} imageData - the data whose copy must be updated on the
+ * registry.
+ */
+function _updateImageDataRegistry(imgSrc, imageData) {
+    const imageDataCopy = Object.assign({}, imageData);
+    registry.category("image.data").add(imgSrc, imageDataCopy, { force: true });
+}
+/**
+ * Returns a proxy of the data associated to imgEl. If a data present in
+ * 'DATASET_IMAGE_OPTIONS' is updated on the proxy, it updates the dataset of
+ * the image accordingly.
+ *
+ * @param {HTMLImageElement} imgEl
+ * @param {Object} preImageData - default data.
+ * @returns {Object} A proxy of the data associated to imgEl.
+ */
+function _getImageData(imgEl, preImageData = {}) {
+    const imgSrc = imgEl.getAttribute("src");
+    const imageData = registry.category("image.data").get(imgSrc, {});
+
+    const handlerImageData = {
+        set(obj, optionName, optionValue) {
+            if (_isDatasetImageOption(optionName)) {
+                // If a "data-attribute option" is modified, also modify the
+                // option in the dataset.
+                imgEl.dataset[optionName] = optionValue;
+            }
+            return Reflect.set(...arguments);
+        },
+        deleteProperty(imageData, optionName) {
+            if (_isDatasetImageOption(optionName)) {
+                // If a "data-attribute option" is removed, remove it also from
+                // the dataset.
+                delete imgEl.dataset[optionName];
+            }
+            return Reflect.deleteProperty(...arguments);
+        },
+    };
+    return new Proxy(Object.assign({}, preImageData, imageData), handlerImageData);
+}
+/**
+ * Converts a snake case string to camel case.
+ *
+ * @param {string} snakeString - the snake case string to convert to camel case.
+ * @returns {string} The camel case version of snakeString.
+ */
+function _convertSnakeToCamelString(snakeString) {
+    return snakeString.replace(/(_[a-z])/g, (group) => group.replace("_", "").toUpperCase());
+}
+/**
+ * Converts a camel case string to snake case.
+ *
+ * @param {string} camelString - the camel case string to convert to snake case.
+ * @returns {string} The snake case version of camelString.
+ */
+function _convertCamelToSnakeString(camelString) {
+    return camelString.replace(/([a-z])([A-Z])/g, "$1_$2").toLowerCase();
+}
+/**
+ * Converts the keys of an object from snake to camel cases if snakeToCamel is
+ * set to 'true' or from camel to snake cases otherwise.
+ *
+ * @param {Object} objectToConvert - the object whose keys have to be converted.
+ * @param {boolean} snakeToCamel - true if the keys of objectToConvert are snake
+ * cases and they have to be converted to camel cases and false if they are
+ * camel cases and they have to be converted to snake cases.
+ * @returns {Object} A copy of objectToConvert whose keys have been converted.
+ */
+function _convertSnakeToCamelOrCamelToSnakeObject(objectToConvert, snakeToCamel) {
```

Comment:
Is that actually needed? Feels weird. As nothing will be saved in the DOM, I don't think there is any need. Treat all options as things saved in python and thus the image data in the JS can be handling those values like this: `someImageData['original_id']` etc.

Reply:
Ok :+1: Its goal was indeed to be able to call the options in snake case from the py and in camel case from the js but I will adapt the code and remove this function to use `someImageData['original_id']` from the JS :+1: